### PR TITLE
Cdk support dev integ test deploy

### PR DESCRIPTION
**Pull request by hawflau**
#### Checklist

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [ ] Write unit tests
- [ ] Write/update functional tests
- [ ] Write/update integration tests
- [ ] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

**Review by wchengru — APPROVED** [review]

File: tests/integration/deploy/test_deploy_command_cdk.py [hunk 1]
```diff
@@ -0,0 +1,742 @@
...
+            if stack_name != SAM_CLI_STACK_NAME:
+                self.cf_client.delete_stack(StackName=stack_name)
+        super().tearDown()
+
+    def _method_to_stack_name(self, method_name):
+        """Method expects method name which can be a full path. Eg: test.integration.test_deploy_command.method_name"""
+        method_name = method_name.split(".")[-1]
+        return f"{method_name.replace('_', '-')}-{CFN_PYTHON_VERSION_SUFFIX}"
+
+    def _replace_cloud_assembly_stack_name(self, manifest_path, original_stack_name, new_stack_name):
```
Comment (mildaniel) [inline tests/integration/deploy/test_deploy_command_cdk.py:69]: Question: What's the purpose of this function/is it used in these tests?

File: samcli/lib/iac/interface.py [hunk 2]
```diff
@@ -739,6 +739,12 @@ def stacks(self) -> List[Stack]:
     def stacks(self, stacks: List[Stack]) -> None:
         self._stacks = stacks
 
+    @property
+    def default_stack(self) -> Optional[Stack]:
+        if len(self.stacks) > 0:
```
Comment (mildaniel) [inline samcli/lib/iac/interface.py:744]: Is there a reason for checking the length of `self.stacks` but returning a stack from `self._stacks`?

File: tests/integration/deploy/test_deploy_command_cdk.py (hunk 1 above)
Comment (hawflau) [reply tests/integration/deploy/test_deploy_command_cdk.py:69]: sorry, it's not needed anymore. Will remove it.

File: samcli/lib/iac/interface.py (hunk 2 above)
Comment (hawflau) [reply samcli/lib/iac/interface.py:744]: It's to ensure `self.stacks` contains at least one stack. Otherwise, it returns `None`.

Comment (mildaniel) [reply samcli/lib/iac/interface.py:744]: I just meant why the check is for the public member `self.stacks` but the return is a value of the protected member `self._stacks`. Just for my understanding.

Comment (hawflau) [reply samcli/lib/iac/interface.py:744]: ah got it. They are the same but I agree that it should be either both internal or both public. I'll update it

Comment (hawflau) [reply samcli/lib/iac/interface.py]: updated

**Review by mildaniel — APPROVED** [review]

**Pull request merged**